client_app/src/Components/IndividualShelter.js

```diff
@@ -38,14 +38,15 @@ const IndividualShelter = (props) => {
       {value}
     </button>
   ));
+  const { v4: uuidv4 } = require('uuid');
 
   function addShift() {
     if (props.addShiftFunction) {
       let id = shelter.id;
       let start = startTime.getTime();
       let end = endTime.getTime();
       let shift = {
-        code: `${id}-${start}-${end}`,
+        code: `${uuidv4()}-${id}`,
```

Comment:
It seems to me that the unique code should be generated by the server/database, rather than the client side.